backend/inspirehep/records/links.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+from flask import current_app, request, url_for
+from invenio_records_rest import current_records_rest
+
+from inspirehep.pidstore.api import PidStoreBase
+
+
+def inspire_detail_links_factory(pid, record=None, record_hit=None, *args, **kwargs):
+    links = {}
+    endpoint = find_record_endpoint(pid, record_hit)
+    endpoint_item = "invenio_records_rest.{0}_item".format(endpoint)
+    self = url_for(endpoint_item, pid_value=pid.pid_value, _external=True)
+
+    serializers_endpoints = current_app.config[endpoint.upper()].get(
+        "record_serializers_aliases", {}
+    )
+    for serializer_endpoint in serializers_endpoints.keys():
+        links[serializer_endpoint] = f"{self}?format={serializer_endpoint}"
+
+    additional_endpoints = current_app.config.get("ADDITIONAL_LINKS", {}).get(
+        endpoint.upper(), {}
+    )
+    for additional_endpoint_name, additional_endpoint in additional_endpoints.items():
+        links[additional_endpoint_name] = url_for(
+            additional_endpoint, pid_value=pid.pid_value, _external=True
+        )
+
+    return links
+
+
+def inspire_search_links(links):
+    self_url = None
+    if links:
+        self_url = links.get("self")
+    if not self_url:
+        return {}
+    endpoint = request.path[1:]
+    formats = (
+        current_app.config[endpoint.upper()]
+        .get("search_serializers_aliases", {})
+        .keys()
+    )
+    links.update(
+        {format_name: f"{self_url}&format={format_name}" for format_name in formats}
+    )
+
+    return links
+
+
+def find_record_endpoint(pid, record_hit=None, **kwargs):
+    if pid.pid_type != "recid":
+        return current_records_rest.default_endpoint_prefixes[pid.pid_type]
+    elif record_hit and "$schema" in record_hit.get("_source", {}):
```

Comment:
why is this branch needed? is the same function called on search results and on detailed records? AFAICT it's only used for individual records that have a `pid.pid_type` set that could be used.

Reply:
Yes, same function is called for search results and detailed records